README.md

```diff
@@ -1,28 +1,84 @@
-# typescript-pkg-template
+# zod-non-empty-array
 
-Simple Typescript template to quickstart your next project.
+Make zod compatible with fp-ts non-empty constructs.
 
-[![Build Status](https://github.com/causaly/typescript-pkg-template/actions/workflows/ci.yml/badge.svg)](https://github.com/causaly/typescript-pkg-template/actions/workflows/ci.yml) [![npm version](https://img.shields.io/npm/v/typescript-pkg-template.svg?color=0c0)](https://www.npmjs.com/package/typescript-pkg-template)
+[![Build Status](https://github.com/causaly/zod-non-empty-array/actions/workflows/ci.yml/badge.svg)](https://github.com/causaly/zod-non-empty-array/actions/workflows/ci.yml) [![npm version](https://img.shields.io/npm/v/zod-non-empty-array.svg?color=0c0)](https://www.npmjs.com/package/zod-non-empty-array)
 
-## Instructions
+## Installation
 
-- Click the `Use this template` button
-- Update the appropriate metadata in `package.json`, depending on the environments you'll use
-- Update LICENSE
-- Write your code under `/lib`
+```bash
+npm install zod-non-empty-array
+```
 
 #### Requirements
 
 - Node.js v.16+
 
-## Notes
+## Quick start
 
-The package by default exports artifacts for both browser & node environments
+```typescript
+import { z as zod } from 'zod';
+import { identity } from 'fp-ts/lib/function';
+import * as NonEmptyArray from 'fp-ts/NonEmptyArray';
+import { transformNonEmptyArray } from 'zod-non-empty-array';
 
-## Publishing
+// create zod schema
+const schema = zod.object({
+  id: zod.number().int().positive(),
+  title: zod.string(),
+  authors: zod
+    .array(zod.string())
+    // must have at least one author
+    .transform(transformNonEmptyArray),
+});
 
-The repo is using changesets and the accompanying GitHub action to handle versioning, keeping a changelong and publishing to NPM.
-You can read more about the changesets package [here](https://github.com/changesets/changesets).
+type Article = zod.infer<typeof schema>;
+
+// parse some invalid value
+const article = schema.parse({
+  id: 1,
+  title: 'A case-study of the use of fp-ts in a real-world project',
+  authors: ['John Doe', 'Jane Doe'],
+});
+
+// article.authors is now compatible with fp-ts
+const principalAuthor = pipe(article.authors, NonEmptyArray.head);
+```
+
+## Motivation
+
+Zod already supports [non-empty array validation](https://zod.dev/?id=nonempty). However, the generated type is incompatible with [fp-ts](https://gcanti.github.io/fp-ts/).
+
+Specifically, fp-ts defines `NonEmptyArray` as follows...
+
+```typescript
+interface NonEmptyArrayAlt<T> extends Array<T> {
```

Comment:
[question] Nit, but does `fp-ts` actually has the `Alt` suffix, or is this something ours? 

```suggestion
interface NonEmptyArray<T> extends Array<T> {
```

Reply:
This is mistake. Thanks for noticing.

Reply:
Fixed [a5c2e06](https://github.com/causaly/zod-non-empty-array/pull/1/commits/a5c2e0681a5601114394ce98ed03dcd586a5b38b)